internal/service/subscription_handler.go

```diff
@@ -403,59 +428,74 @@ func (h *alarmSubscriptionHandler) deleteToSubscriptionMap(key string) {
 	delete(*h.subscriptionMap, key)
 }
 
-func (h *alarmSubscriptionHandler) assignSubscriptionMap(newMap map[string]data.Object) {
-	h.subscritionMapMemoryLock.Lock()
-	defer h.subscritionMapMemoryLock.Unlock()
+func (h *SubscriptionHandler) assignSubscriptionMap(newMap map[string]data.Object) {
+	h.subscriptionMapLock.Lock()
+	defer h.subscriptionMapLock.Unlock()
 	h.subscriptionMap = &newMap
 }
 
-func (h *alarmSubscriptionHandler) getSubcriptionId() (subId string) {
+func (h *SubscriptionHandler) getSubcriptionId() (subId string) {
 	subId = uuid.New().String()
 	return
 }
 
-func (h *alarmSubscriptionHandler) encodeSubId(ctx context.Context,
+func (h *SubscriptionHandler) encodeSubId(
 	subId string, input data.Object) (output data.Object, err error) {
-	//get consumer name, subscriptions
-	err = h.jqTool.Evaluate(
-		`{
-			"alarmSubscriptionId": $alarmSubId,
-			"consumerSubscriptionId": .consumerSubscriptionId,
-			"callback": .callback,
-			"filter": .filter
-		}`,
-		input, &output,
-		jq.String("$alarmSubId", subId),
-	)
+
+	err = nil
+
+	if h.persistStore.Name == AlarmConfigMapName {
+		// get consumer name, subscriptions
+		err = h.jqTool.Evaluate(
+			`{
+				"alarmSubscriptionId": $alarmSubId,
+				"consumerSubscriptionId": .consumerSubscriptionId,
+				"callback": .callback,
+				"filter": .filter
+			}`,
+			input, &output,
+			jq.String("$alarmSubId", subId),
+		)
+	} else if h.persistStore.Name == InfraInventoryConfigMapName {
+		// get consumer name, subscriptions
+		err = h.jqTool.Evaluate(
+			`{
+				"subscriptionId": $subId,
```

Comment:
Just to keep the convention, maybe we can rename it to something like `$resourceSubId`

Reply:
I  wanted to keep it in line with the O-RAN standard (where it's `subscriptionId`) and the alarm subscription case that's using `alarmSubscriptionId`, just as in the standard.
Please let me know if you agree.

Reply:
Oh, I was actually referring to the name of the variable (`$subId`). It just looked a bit weird at first glance since it's generic compared to `$alarmSubId`. But feel free to ignore, it makes better sense now.